yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher.java

```diff
@@ -179,19 +174,26 @@ public void launch(@Nonnull SlaveComputer computer, TaskListener listener) throw
             throw new NullPointerException("Node can't be null");
         }
         final DockerSlaveTemplate dockerSlaveTemplate = node.getDockerSlaveTemplate();
+        final DockerComputerJNLPLauncher launcher = (DockerComputerJNLPLauncher) dockerSlaveTemplate.getLauncher();
+
+        final String rootUrl = launcher.getJenkinsUrl(Jenkins.getActiveInstance().getRootUrl());
+//        Objects.requireNonNull(rootUrl, "Jenkins root url is not specified!");
+        if (isNull(rootUrl)) {
+            throw new NullPointerException("Jenkins root url is not specified!");
+        }
 
         // exec jnlp connection in running container
         // TODO implement PID 1 replacement
         String startCmd =
                 "cat << EOF > /tmp/config.sh.tmp && cd /tmp && mv config.sh.tmp config.sh\n" +
                         "JENKINS_URL=\"" + rootUrl + NL +
-                        "JENKINS_USER=\"" + getUser() + NL +
+                        "JENKINS_USER=\"" + launcher.getUser() + NL +
                         "JENKINS_HOME=\"" + dockerSlaveTemplate.getRemoteFs() + NL +
                         "COMPUTER_URL=\"" + dockerComputer.getUrl() + NL +
                         "COMPUTER_SECRET=\"" + dockerComputer.getJnlpMac() + NL +
-                        "JAVA_OPTS=\"" + getJvmOpts() + NL +
-                        "SLAVE_OPTS=\"" + getSlaveOpts() + NL +
-                        "NO_CERTIFICATE_CHECK=\"" + isNoCertificateCheck() + NL +
+                        "JAVA_OPTS=\"" + launcher.getJvmOpts() + NL +
```

Comment:
may be there is better to fix it. but this is just empty. So, I needed to get it from DockerComputer.